### PR TITLE
Accs1472

### PR DESCRIPTION
**Pull request by danInCode**
Fix to the following problem: Failed to export the mesh whenever the WriteToDMTFile or WriteToSTLFile was called several times in a row.

### CODE REVIEW

File: Delcam.ProductInterface.PowerSHAPE.Test/DatabaseEntity Tests/Entity Tests/MeshTest.cs [hunk 1]
```diff
@@ -249,6 +250,52 @@ public void MinimumDistanceMeshCompCurve()
...
+        public void WhenWriteToDMTFile_CheckMeshDidNotChange()
+        {
+            // Import mesh
+            _powerSHAPE.Reset();
+            var importedMesh =_powerSHAPE.ActiveModel.Meshes.CreateMeshFromFile(new File(TestFiles.SINGLE_MESH));
+            var numberOfTriangles = importedMesh.NumberOfTriangles;
+            var boundingBox = importedMesh.BoundingBox;
+
+            // Export mesh 
+            var output = File.CreateTemporaryFile("dmt");
```
Comment (lukeedw) [inline Delcam.ProductInterface.PowerSHAPE.Test/DatabaseEntity Tests/Entity Tests/MeshTest.cs:267]: You need to clean up this temporary file.  Preferably do it before any Asserts so that the file is cleaned up even if the test fails.

File: Delcam.ProductInterface.PowerSHAPE.Test/DatabaseEntity Tests/Entity Tests/MeshTest.cs [hunk 2]
```diff
@@ -249,6 +250,52 @@ public void MinimumDistanceMeshCompCurve()
...
+        public void WhenWriteToSTLFile_CheckMeshDidNotChange()
+        {
+            // Import mesh
+            _powerSHAPE.Reset();
+            var importedMesh = _powerSHAPE.ActiveModel.Meshes.CreateMeshFromFile(new File(TestFiles.SINGLE_MESH));
+            var numberOfTriangles = importedMesh.NumberOfTriangles;
+            var boundingBox = importedMesh.BoundingBox;
+
+            // Export mesh 
+            var output = File.CreateTemporaryFile("stl");
```
Comment (lukeedw) [inline Delcam.ProductInterface.PowerSHAPE.Test/DatabaseEntity Tests/Entity Tests/MeshTest.cs:291]: You need to clean up this temporary file too.

File: Delcam.ProductInterface.PowerSHAPE.Test/DatabaseEntity Tests/Entity Tests/MeshTest.cs [hunk 3]
```diff
@@ -249,6 +250,52 @@ public void MinimumDistanceMeshCompCurve()
...
+            _powerSHAPE.Reset();
+            var importedMesh =_powerSHAPE.ActiveModel.Meshes.CreateMeshFromFile(new File(TestFiles.SINGLE_MESH));
+            var numberOfTriangles = importedMesh.NumberOfTriangles;
+            var boundingBox = importedMesh.BoundingBox;
+
+            // Export mesh 
+            var output = File.CreateTemporaryFile("dmt");
+            importedMesh.WriteToDMTFile(output);
+
+            // Test the distance
```
Comment (lukeedw) [inline Delcam.ProductInterface.PowerSHAPE.Test/DatabaseEntity Tests/Entity Tests/MeshTest.cs]: Distance?  Test what distance?

File: Delcam.ProductInterface.PowerSHAPE.Test/DatabaseEntity Tests/Entity Tests/MeshTest.cs [hunk 4]
```diff
@@ -249,6 +250,52 @@ public void MinimumDistanceMeshCompCurve()
...
+            _powerSHAPE.Reset();
+            var importedMesh = _powerSHAPE.ActiveModel.Meshes.CreateMeshFromFile(new File(TestFiles.SINGLE_MESH));
+            var numberOfTriangles = importedMesh.NumberOfTriangles;
+            var boundingBox = importedMesh.BoundingBox;
+
+            // Export mesh 
+            var output = File.CreateTemporaryFile("stl");
+            importedMesh.WriteToSTLFile(output);
+
+            // Test the distance
```
Comment (lukeedw) [inline Delcam.ProductInterface.PowerSHAPE.Test/DatabaseEntity Tests/Entity Tests/MeshTest.cs]: Difference?

File: Delcam.ProductInterface.PowerSHAPE.Test/Delcam.ProductInterface.PowerSHAPE.Test.csproj [hunk 5]
```diff
@@ -261,4 +265,11 @@
     </ProjectReference>
   </ItemGroup>
   <Import Project="$(MSBuildBinPath)\Microsoft.CSharp.targets" />
+  <Import Project="..\packages\MSBuildTasks.1.5.0.235\build\MSBuildTasks.targets" Condition="Exists('..\packages\MSBuildTasks.1.5.0.235\build\MSBuildTasks.targets')" />
```
Comment (lukeedw) [inline Delcam.ProductInterface.PowerSHAPE.Test/Delcam.ProductInterface.PowerSHAPE.Test.csproj]: Looks like your PC has added something unnecessary here?

File: Delcam.ProductInterface.PowerSHAPE.Test/packages.config [hunk 6]
```diff
@@ -1,5 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
+  <package id="MSBuildTasks" version="1.5.0.235" targetFramework="net45" developmentDependency="true" />
```
Comment (lukeedw) [inline Delcam.ProductInterface.PowerSHAPE.Test/packages.config]: What is this?

File: Delcam.ProductInterface.PowerShape/Entities/Mesh/PSMesh.cs [hunk 7]
```diff
@@ -9,6 +9,7 @@
 
 using System;
 using System.Collections.Generic;
+using System.Diagnostics;
```
Comment (lukeedw) [inline Delcam.ProductInterface.PowerShape/Entities/Mesh/PSMesh.cs]: Why was this added?

**Review by lukeedw — CHANGES_REQUESTED** [review]
A couple of code improvements but also some odd changes that may be a configuration of your PC.

File: Delcam.ProductInterface.PowerSHAPE.Test/DatabaseEntity Tests/Entity Tests/MeshTest.cs (hunk 1 above)
Comment (danInCode) [reply Delcam.ProductInterface.PowerSHAPE.Test/DatabaseEntity Tests/Entity Tests/MeshTest.cs:267]: Done

File: Delcam.ProductInterface.PowerSHAPE.Test/DatabaseEntity Tests/Entity Tests/MeshTest.cs (hunk 2 above)
Comment (danInCode) [reply Delcam.ProductInterface.PowerSHAPE.Test/DatabaseEntity Tests/Entity Tests/MeshTest.cs:291]: Done

File: Delcam.ProductInterface.PowerSHAPE.Test/DatabaseEntity Tests/Entity Tests/MeshTest.cs (hunk 3 above)
Comment (danInCode) [reply Delcam.ProductInterface.PowerSHAPE.Test/DatabaseEntity Tests/Entity Tests/MeshTest.cs]: Done. Changed the comment

File: Delcam.ProductInterface.PowerSHAPE.Test/DatabaseEntity Tests/Entity Tests/MeshTest.cs (hunk 4 above)
Comment (danInCode) [reply Delcam.ProductInterface.PowerSHAPE.Test/DatabaseEntity Tests/Entity Tests/MeshTest.cs]: Done. Changed the comment

File: Delcam.ProductInterface.PowerSHAPE.Test/Delcam.ProductInterface.PowerSHAPE.Test.csproj (hunk 5 above)
Comment (danInCode) [reply Delcam.ProductInterface.PowerSHAPE.Test/Delcam.ProductInterface.PowerSHAPE.Test.csproj]: Removed

File: Delcam.ProductInterface.PowerSHAPE.Test/packages.config (hunk 6 above)
Comment (danInCode) [reply Delcam.ProductInterface.PowerSHAPE.Test/packages.config]: Removed

File: Delcam.ProductInterface.PowerShape/Entities/Mesh/PSMesh.cs (hunk 7 above)
Comment (danInCode) [reply Delcam.ProductInterface.PowerShape/Entities/Mesh/PSMesh.cs]: Removed

**Comment by danInCode** [comment]
Luke, I have implemented all the changes. Please let me know if I can now merge the PR.

**Review by lukeedw — APPROVED** [review]
